### PR TITLE
fix: Avoid edge case with 409 causing more SSE requests

### PR DESCRIPTION
**Pull request by msfstef**
Fix test flake (see [this run](https://github.com/electric-sql/electric/actions/runs/19533935381/job/55923347981) for example) caused by 409s for the shape resetting the fallback from SSE to long polling.

Not sure if 409s _should_ reset this fallback but it does, and the test should account for it as we occasionally see these 409s from the tests overlapping each other.

### CODE REVIEW

File: packages/typescript-client/test/client.test.ts [hunk 1]
```diff
@@ -2275,6 +2276,7 @@ it(
   async ({ issuesTableUrl, aborter }) => {
     let requestCount = 0
     let sseRequestCount = 0
+    let initialHandle: string | null = null
```
Comment (kevin-dp) [inline packages/typescript-client/test/client.test.ts]: Why `string | null` and not `string | undefined`, then we could shorten it to `let initialHandle: string | undefined`

File: packages/typescript-client/test/client.test.ts [hunk 2]
```diff
@@ -2314,7 +2319,7 @@ it(
               status: 200,
               headers: new Headers({
                 'electric-offset': `0_0`,
-                'electric-handle': `test-handle`,
+                'electric-handle': reqHandle!,
```
Comment (kevin-dp) [inline packages/typescript-client/test/client.test.ts:2322]: nit: i think the `!` operator would be more appropriate in the definition of `const reqHandle` rather than here.

Comment (msfstef) [reply packages/typescript-client/test/client.test.ts:2322]: I disagree - there will be requests made with the fetch client that do not include a handle (initial request)
If it is an SSE live request, it is guaranteed to have a handle, but I'd rather not do a nested check

File: packages/typescript-client/test/client.test.ts (hunk 1 above)
Comment (msfstef) [reply packages/typescript-client/test/client.test.ts]: because I'm setting `intialHandle ??= reqHandle`, and `reqHandle` can be `string | null` at that point.

that being said I assert it later one so I could just assert it here.

Comment (kevin-dp) [reply packages/typescript-client/test/client.test.ts]: > because I'm setting `intialHandle ??= reqHandle`, and `reqHandle` can be `string | null` at that point.
> 
> that being said I assert it later one so I could just assert it here.

Ah ok, leave it like this than! :)

File: packages/typescript-client/test/client.test.ts (hunk 2 above)
Comment (kevin-dp) [reply packages/typescript-client/test/client.test.ts:2322]: Oh ok missed that, judging by the code i though it would always be present.
